autoextract_poet/items.py

```diff
@@ -2,17 +2,28 @@
 
 import attr
 
+from autoextract_poet.util import attr_prepare
+
 
 @attr.s(auto_attribs=True, slots=True)
 class Item:
 
     @classmethod
     def from_dict(cls, item: Optional[Dict]):
+        """
+        Read an item from a dictionary, ignoring unknown attributes for
+        backwards compatibility
+        """
+        item = attr_prepare(item, cls)
         return cls(**item) if item else None  # type: ignore
 
     @classmethod
-    def from_list(cls, items: List[Dict]):
-        return [cls.from_dict(item) for item in items if item]
+    def from_list(cls, items: Optional[List[Dict]]) -> List:
+        """
+        Read items from a list, ignoring unknown attributes for
+        backwards compatibility, and filtering out ``None`` attributes
+        """
+        return [cls.from_dict(item) for item in items or [] if item]
```

Comment:
I think we shouldn't have the `if item` here.
The `None` condition is already handled in `from_dict`.
Then, if I call `from_list([None, data])` I'll get `[data]` and they differ in length

Reply:
Makes sense. Done in https://github.com/scrapinghub/autoextract-poet/pull/9/commits/cc9b485fb6707f600612d9b05388206393e0a4fb